### PR TITLE
Some general cleanup/fixes for Arc dashboards

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
* Add missing README section
* Better error handling in dashboards, prompting for reconnections where appropriate and not showing loading spinners indefinitely
* Add disposable store for dashboard pages to fix resource leaking
* Add refresh buttons to Controller/MIAA dashboards
* Move model refresh calls to dashboard to decouple that from the tree nodes and ensure proper ordering

@bergeron There's definitely going to be some merge conflicts with your stuff (like around the disposables) so go ahead and merge in your changes and then I'll update this PR at that point. Leaving this as a draft until then.

### CODE REVIEW

File: extensions/arc/src/ui/components/dashboardPage.ts [hunk 1]
```diff
@@ -4,12 +4,21 @@
...
+import * as vscode from 'vscode';
 import { InitializingComponent } from './initializingComponent';
 
 export abstract class DashboardPage extends InitializingComponent {
 
+	protected disposables: vscode.Disposable[] = [];
+
 	constructor(protected modelView: azdata.ModelView) {
 		super();
+		this.disposables.push(modelView.onClosed(() => {
```
Comment (bergeron) [inline extensions/arc/src/ui/components/dashboardPage.ts:16]: Nice one, I missed disposing the onClosed listener

File: extensions/arc/src/ui/dashboards/miaa/miaaDashboardOverviewPage.ts [hunk 2]
```diff
@@ -40,14 +40,16 @@ export class MiaaDashboardOverviewPage extends DashboardPage {
...
+				this.eventuallyRunOnInitialized(() => {
+					this.handleRegistrationsUpdated().catch(e => console.log(e));
+				});
+			}),
+			this._controllerModel.onEndpointsUpdated(endpoints => this.eventuallyRunOnInitialized(() => this.handleEndpointsUpdated(endpoints))),
+			this._miaaModel.onStatusUpdated(status => this.eventuallyRunOnInitialized(() => this.handleMiaaStatusUpdated(status))),
+			this._miaaModel.onDatabasesUpdated(databases => this.eventuallyRunOnInitialized(() => this.handleDatabasesUpdated(databases)))
+		);
 
 		this.refresh().catch(e => {
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/miaa/miaaDashboardOverviewPage.ts:54]: MiaaDashboard already refreshes the miaaModel so I expect these refreshes in the page constructors to go away.

Comment (Charles-Gagnon) [reply extensions/arc/src/ui/dashboards/miaa/miaaDashboardOverviewPage.ts]: Good catch, fixed to only have the dashboard refresh when opened

File: extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts [hunk 3]
```diff
@@ -24,11 +23,6 @@ export class PostgresResourceHealthPage extends DashboardPage {
 		modelView.onClosed(() => {
```
Comment (bergeron) [inline extensions/arc/src/ui/dashboards/postgres/postgresResourceHealthPage.ts:23]: this.disposables.push this one too?  Thanks for fixing Postgres by the way.

**Pull request merged**